libraries/node-core-library/src/Async.ts

```diff
@@ -154,4 +154,27 @@ export class Async {
       setTimeout(resolve, ms);
     });
   }
+
+  /**
+   * Executes an async function and optionally retries it if it fails.
+   */
+  public static async runWithRetriesAsync<TResult>(
+    fn: () => Promise<TResult> | TResult,
```

Comment:
```suggestion
    action: () => Promise<TResult> | TResult,
```
`fn` is not an English word.